model/src/cfd.rs

```diff
@@ -3103,6 +3134,114 @@ mod tests {
+    }
+
+    #[test]
+    fn give_order_creation_timestamp_not_outdated_then_order_not_outdated() {
+        let creation_timestamp = Timestamp::now();
+        let order = Order::dummy_short().with_creation_timestamp(creation_timestamp);
+
+        let now =
+            OffsetDateTime::now_utc() + Duration::seconds(Order::OUTDATED_AFTER_MINS * 60 - 1);
+
+        assert!(!order.is_creation_timestamp_outdated(now))
+    }
+
+    #[test]
+    fn given_oracle_event_id_is_24h_in_the_future_then_sane_to_take() {
+        // --|---------|---------|----------------------------------|--> time
+        //   -1h       |         +1h                                24h
+        // --|---------|<--------|--------------------------------->|--
+        //             now
+
+        let order = Order::dummy_short().with_oracle_event_id(BitMexPriceEventId::with_20_digits(
+            datetime!(2021-11-19 10:00:00).assume_utc(),
+        ));
+
+        let sane =
+            order.is_oracle_event_id_sane_to_take(datetime!(2021-11-18 10:00:00).assume_utc());
+        assert!(sane)
+    }
+
+    #[test]
+    fn given_oracle_event_id_is_23h_in_the_future_then_sane_to_take() {
+        // --|---------|---------|----------------------------------|--> time
+        //   -1h       |         +1h                                24h
+        // --|---------|<--------|--------------------------------->|--
+        //                       now
+
+        let order = Order::dummy_short().with_oracle_event_id(BitMexPriceEventId::with_20_digits(
+            datetime!(2021-11-19 10:00:00).assume_utc(),
+        ));
+
+        let sane =
+            order.is_oracle_event_id_sane_to_take(datetime!(2021-11-18 11:00:00).assume_utc());
+        assert!(sane, "the oracle event id is outdated")
+    }
+
+    #[test]
+    fn given_oracle_event_id_is_25h_in_the_future_then_sane_to_take() {
+        // --|---------|---------|----------------------------------|--> time
+        //   -1h       |         +1h                                24h
+        // --|---------|<--------|--------------------------------->|--
+        //   now
+
+        let order = Order::dummy_short().with_oracle_event_id(BitMexPriceEventId::with_20_digits(
+            datetime!(2021-11-19 10:00:00).assume_utc(),
+        ));
+
+        let sane =
+            order.is_oracle_event_id_sane_to_take(datetime!(2021-11-18 09:00:00).assume_utc());
+        assert!(sane, "the oracle event id is to far in the future")
+    }
+
+    #[test]
+    fn given_oracle_event_id_is_more_than_25h_in_the_future_then_not_sane_to_take() {
+        // --|---------|---------|----------------------------------|--> time
+        //   -1h1s     |         +1h                                24h
+        // --|---------|<--------|--------------------------------->|--
+        //   now
+
+        // Note the maker cannot go back in time, if an offer was created with a timestamp of 10:00
+        // it must have been at least 9:00:00
```

Comment:
🤔 I'm not sure this comment is necessary here. These tests are for a function called by the taker. The taker doesn't need to know how the maker comes up with oracle event IDs. They just need to verify that what they get from the maker is acceptable to them.

Reply:
I removed it :)